ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,295 @@
+                _add_health_check_to_elb(lb, health_check)
+    else:
+        raise NonRecoverableError(
+            'Health Checks not defined properly in inputs. '
+            'Health Checks should be a list.')
+
+
+def _add_instance_to_elb_list_in_properties(instance_id):
+
+    if 'instance_list' not in ctx.target.instance.runtime_properties.keys():
+        ctx.target.instance.runtime_properties['instance_list'] = []
+
+    ctx.target.instance.runtime_properties['instance_list'].append(instance_id)
+
+
+def _remove_instance_from_elb_list_in_properties(instance_id):
+    if instance_id in ctx.target.instance.runtime_properties['instance_list']:
+        ctx.target.instance.runtime_properties['instance_list'].remove(
+            instance_id)
+
+
+@operation
+def remove_instance_from_elb(**_):
+
+    ctx.logger.info('Attempting to remove_instance_from_elb.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    instance_list = [instance_id]
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.deregister_instances(instance_list)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Instance not removed from Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} removed from Load Balancer {1}.'
+        .format(instance_id, elb_name))
+    _remove_instance_from_elb_list_in_properties(instance_id)
+
+
+@operation
+def add_instance_to_elb(**_):
+
+    ctx.logger.info('Attempting to add_instance_to_elb.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    lb = _get_existing_elb(elb_name)
+
+    try:
+        lb.register_instances(instance_id)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+            raise NonRecoverableError('Instance not added to Load Balancer. '
+                                      '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} added to Load Balancer {1}.'
+        .format(instance_id, elb_name))
+
+    _add_instance_to_elb_list_in_properties(instance_id)
+
+
+def _add_health_check_to_elb(elb, health_check):
+
+    hc = _create_health_check(health_check)
+
+    try:
+        elb.configure_health_check(hc)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Health check not added to Load Balancer. '
```

Comment:
boto has its own retry mechanism, so my assumption (yes, assumption is the mother of all...) is that it takes care of connectivity issues and If it decided to fail, there is no point in me trying again. But then again, you guys are the experts, What do you think ?